### PR TITLE
Document OkHttp callback for Android

### LINKED ISSUE #3630: Document OkHttp callback for Android
## Issue

https://github.com/getsentry/sentry-java/pull/1478
A new callback to customize the OkHttp integration is added hence we need to document it

## Description

Document how to add/use the callback https://docs.sentry.io/platforms/android/configuration/integrations/okhttp/

### Suggested Solution

Document how to add/use the callback https://docs.sentry.io/platforms/android/configuration/integrations/okhttp/

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
Fixes #3630

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
cc @marandaneto

File: src/platforms/android/configuration/integrations/okhttp.mdx [hunk 1]
```diff
@@ -110,3 +110,61 @@ Learn more about manually capturing an error or message, in our <PlatformLink to
 </Note>
 
 To view and resolve the recorded message, log into [sentry.io](https://sentry.io) and open your project. Clicking on the error's title will open a page where you can see detailed information and mark it as resolved.
+
+## Customize recorded Span
```
Comment (imatwawana) [inline src/platforms/android/configuration/integrations/okhttp.mdx:114]:
```suggestion
## Customize the Recorded Span
```

File: src/platforms/android/configuration/integrations/okhttp.mdx [hunk 2]
```diff
@@ -110,3 +110,61 @@ Learn more about manually capturing an error or message, in our <PlatformLink to
...
+
+Captured Span can be customized or dropped with a `BeforeSpanCallback`:
```
Comment (imatwawana) [inline src/platforms/android/configuration/integrations/okhttp.mdx:116]:
```suggestion
The captured span can be customized or dropped with a `BeforeSpanCallback`:
```

**Review by imatwawana — APPROVED** [review]
Minor wording edits

File: src/platforms/android/configuration/integrations/okhttp.mdx [hunk 3]
```diff
@@ -110,3 +110,61 @@ Learn more about manually capturing an error or message, in our <PlatformLink to
 </Note>
 
 To view and resolve the recorded message, log into [sentry.io](https://sentry.io) and open your project. Clicking on the error's title will open a page where you can see detailed information and mark it as resolved.
+
+## Customize the Recorded Span
```
Comment (marandaneto) [inline src/platforms/android/configuration/integrations/okhttp.mdx:114]: let's add a note that this is available thru the beta version, similar to https://github.com/getsentry/sentry-docs/blame/a8ce23144605efb58f44e2ccbafed4f41d2d8aef/src/platforms/java/guides/spring/advanced-usage.mdx#L36-L40

**Review by marandaneto — APPROVED** [review]

File: src/platforms/android/configuration/integrations/okhttp.mdx (hunk 3 above)
Comment (marandaneto) [inline src/platforms/android/configuration/integrations/okhttp.mdx:114]:
```suggestion
## Customize the Recorded Span

<Alert level="warning" title="Note">

`BeforeSpanCallback` is available in 5.0 beta version of the Sentry Java SDK.

</Alert>
```

**Pull request merged**